src/lib.rs

```diff
@@ -439,6 +579,45 @@ impl Navigator {
         }
     }
 
+    /// Reads all sensors and stores on a single structure.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use navigator_rs::{Navigator, SensorData};
+    /// use std::thread::sleep;
+    /// use std::time::Duration;
+    ///
+    /// let mut nav = Navigator::new();
+    /// nav.init();
+    ///
+    /// loop {
+    ///     let sensor_data: SensorData = nav.read_all();
+    ///     println!("---------------reading---------------");
+    ///     println!(
+    ///         "ADC values: {}, {}, {}, {}",
+    ///         sensor_data.adc.channel[0],
+    ///         sensor_data.adc.channel[1],
+    ///         sensor_data.adc.channel[2],
+    ///         sensor_data.adc.channel[3],
+    ///     );
+    ///     println!("PRES   :   P: {}", sensor_data.pressure);
+    ///     println!("TEMP   :   T: {}", sensor_data.temperature);
+    ///     println!(
+    ///         "ACCE   :   X: {}, Y: {}, Z: {}",
+    ///         sensor_data.accelerometer.x, sensor_data.accelerometer.y, sensor_data.accelerometer.z
+    ///     );
+    ///     println!(
+    ///         "GYRO  :   X: {}, Y: {}, Z: {}",
+    ///         sensor_data.gyro.x, sensor_data.gyro.y, sensor_data.gyro.z
+    ///     );
+    ///     println!(
+    ///         "MAGN   :   X: {}, Y: {}, Z: {}",
+    ///         sensor_data.magnetometer.x, sensor_data.magnetometer.y, sensor_data.magnetometer.z
+    ///     );
```

Comment:
can you add units here also? :D

Reply:
changed the example to print structures directly:
    ///     println!("---------------reading---------------");
    ///     let sensor_data: SensorData = nav.read_all();
    ///     println!("{:#?}", sensor_data);
    ///     sleep(Duration::from_millis(1000));
    
  kept an second example acessing this structure:
  
    ///     let sensor_data: SensorData = nav.read_all();
    ///     println!("---------------Reading ADC Channels---------------");
    ///     println!(
    ///         "ADC values: {}, {}, {}, {} [V]",
    ///         sensor_data.adc.channel[0],
    ///         sensor_data.adc.channel[1],
    ///         sensor_data.adc.channel[2],
    ///         sensor_data.adc.channel[3],
    ///         );
    ///     sleep(Duration::from_millis(1000));